### PR TITLE
[K3.1] Crypsis: Online Status Crypsis feature

### PR DESCRIPTION
**Pull request by 810**
This is the new online status feature created by Josh. But updated to lasted version, + fixes some issue's

The other feature pr, needed merged before this pr can be pulled.

### CODE REVIEW

File: components/com_kunena/site/controllers/user.php [hunk 1]
```diff
@@ -331,6 +331,41 @@ function logout() {
 		$this->setRedirectBack();
 	}
 
+	public function status() {
```
Comment (xillibit) [inline components/com_kunena/site/controllers/user.php]: This new code added doesn't follow Joomla! coding standards

File: components/com_kunena/site/template/crypsis/layouts/user/item/status/default.php [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+ * @link        http://www.kunena.org
+ **/
+defined('_JEXEC') or die;
+
+$user = isset($this->user) ? $this->user : KunenaUserHelper::getMyself();
+$status = $user->getStatus();
+$status_text = $user->getStatusText();
+$link = $user->getURL();
+
+switch ($status) {
```
Comment (xillibit) [inline components/com_kunena/site/template/crypsis/layouts/user/item/status/default.php]: Same here the code doesn't follwo Joomla! coding standards

File: libraries/kunena/user/helper.php [hunk 3]
```diff
@@ -321,6 +324,31 @@ public static function isOnline($user, $yes = false, $no = 'offline') {
...
+	/**
+	 * Returns the status of a user. If as session exists, we can return the type of status the user set.
+	 *
+	 * @param mixed  $user
+	 *
+	 * @return int
+	 */
+	public static function getStatus($user) {
+		$user = self::get($user);
+		$online = false;
```
Comment (xillibit) [inline libraries/kunena/user/helper.php]: Same here code doesn't follow Joomla! coding standards

**Comment by xillibit** [comment]
There seem to be conflict in the code now

File: libraries/kunena/user/helper.php (hunk 3 above)
Comment (810) [inline libraries/kunena/user/helper.php]: you forgot this one

File: components/com_kunena/site/controllers/user.php [hunk 4]
```diff
@@ -331,6 +331,69 @@ function logout() {
...
+		$me = KunenaUserHelper::getMyself();
+		$me->status_text = $status_text;
+
+		if (!$me->save())
+		{
+			$this->app->enqueueMessage($me->getError(), 'error');
+		}
+		else
+		{
+			$this->app->enqueueMessage(JText::_('Successfully Saved Status Text'));
```
Comment (xillibit) [inline components/com_kunena/site/controllers/user.php]: Missing language string here

**Comment by xillibit** [comment]
I have an issue : Rendering Error in layout Widget/Modal: Property "class" is not defined in C:\wamp\www\joomla3.3.6\components\com_kunena\template\crypsis\layouts\widget\modal\default.php on line 17
Layout was rendered in C:\wamp\www\joomla3.3.6\components\com_kunena\template\crypsis\layouts\widget\login\logout\default.php on line 158

**Comment by 810** [comment]
any suggestion to fix this?

**Comment by xillibit** [comment]
What should be the default value into $this->class here (the file is \components\com_kunena\template\crypsis\layouts\widget\modal\default.php) :  

```php
$class = ' class="modal' .$state . $transition . $this->class.'"';
```

?

**Comment by 810** [comment]
i have removed it, please try now

**Comment by xillibit** [comment]
Others issues noticied : 

```
Rendering Error in layout Widget/Modal: Property "form" is not defined in C:\wamp\www\joomla3.3.6\components\com_kunena\template\crypsis\layouts\widget\modal\default.php on line 22
Layout was rendered in C:\wamp\www\joomla3.3.6\components\com_kunena\template\crypsis\layouts\widget\login\logout\default.php on line 159
```

To fix one above add line 158 ```->set('form', 'statusText-form')``` into \components\com_kunena\template\crypsis\layouts\widget\login\logout\default.php

```
Rendering Error in layout Widget/Label: Property "class" is not defined in C:\wamp\www\joomla3.3.6\components\com_kunena\template\crypsis\layouts\widget\label\link.php on line 17
Layout was rendered in C:\wamp\www\joomla3.3.6\components\com_kunena\template\crypsis\layouts\user\item\status\default.php on line 47
```

**Comment by 810** [comment]
try now

**Comment by xillibit** [comment]
Better but this one isn't fixed : 

```
`Rendering Error in layout Widget/Label: Property "class" is not defined in C:\wamp\www\joomla3.3.6\components\com_kunena\template\crypsis\layouts\widget\label\link.php on line 17
Layout was rendered in C:\wamp\www\joomla3.3.6\components\com_kunena\template\crypsis\layouts\user\item\status\default.php on line 47
```
It's enough to remove the ```$this->class``` i think in \components\com_kunena\template\crypsis\layouts\widget\label\link.php line 17

**Comment by 810** [comment]
ok, removed it everywhere

**Comment by xillibit** [comment]
All is good now :+1:

**Pull request merged**